### PR TITLE
support array of enums in arguments

### PR DESCRIPTION
**Pull request by algonautshant**

### CODE REVIEW

File: src/main/java/com/algorand/sdkutils/listeners/ResponseGenerator.java [hunk 1]
```diff
@@ -38,7 +38,9 @@ public static void main(ResponseGeneratorArgs args, JCommander command) throws E
         }
 
         Publisher pub = new Publisher();
-        ResponseGenerator subscriber = new ResponseGenerator(args, pub);
+        ResponseGenerator subscriber = new ResponseGenerator(args);
+        pub.subscribeAll(subscriber);
```
Comment (winder) [inline src/main/java/com/algorand/sdkutils/listeners/ResponseGenerator.java:42]: Why did you move the publisher out of the ResponseGenerator? Wouldn't the object be best suited to register the calls that it needs?

File: src/main/java/com/algorand/sdkutils/listeners/ResponseGenerator.java [hunk 2]
```diff
@@ -370,7 +383,7 @@ public void terminate() {
                         }
                     }
                 });
-        /*
+
```
Comment (winder) [inline src/main/java/com/algorand/sdkutils/listeners/ResponseGenerator.java:386]: Could you add some comments to these sections based on our conversations? I realize things aren't perfect, so maybe we can save some time next time with a record of the known issues.

File: src/main/java/com/algorand/sdkutils/utils/TypeDef.java [hunk 3]
```diff
@@ -49,7 +47,50 @@ public TypeDef(
...
+     * @param javaTypeName is the generated code type: e.g. List<abc>, or MyEnumClassName
+     * @param rawTypeName is the type name from the spec file
+     * @param type is a loosely defined tag used by generator e.g. enum, array, etc.
+     * @param propertyName is the class/struct member name
+     * @param goPropertyName, is when provided with x-go-name
+     * @param doc is the comments associated with the parameter
+     * @param required indicates if the field is a required field
+     * @param enumValues is passed if the type is an array of enums
+     */
+    public TypeDef(
```
Comment (winder) [inline src/main/java/com/algorand/sdkutils/utils/TypeDef.java:62]: Could you add one extra field to the other constructor instead of making a copy?

**Review by winder — CHANGES_REQUESTED** [review]
A couple of small requests for you. For the core logic I'm pretty much taking your word on it, since the generated SDK code looks good this code must be working as you intended.

File: src/main/java/com/algorand/sdkutils/listeners/ResponseGenerator.java (hunk 1 above)
Comment (algonautshant) [reply src/main/java/com/algorand/sdkutils/listeners/ResponseGenerator.java:42]: The reason is the following: the constructor of an object should construct an object, and the actions on the object should be done outside the constructor. 

Calling a constructor and not doing anything with the object looks weird. It even triggers lint warning for creating an unused variable.
